options/m_option.c

```diff
@@ -1517,14 +1517,15 @@ static char *print_str_list(const m_option_t *opt, const void *src)
 {
     char **lst = NULL;
     char *ret = NULL;
+    const char *sep = opt->priv ? opt->priv : (char[]){OPTION_LIST_SEPARATOR, 0};
```

Comment:
I would suggest simpler approach. Separator should be one char only.

```suggestion
    const char sep = opt->priv ? *opt->priv : OPTION_LIST_SEPARATOR;
```

Reply:
`*opt->priv` would need to be `*(char *)opt->priv` but sure